test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -149,6 +150,23 @@ func (g *Grabber) GrabFromScheduler() (SchedulerMetrics, error) {
 	if g.kubeScheduler == "" {
 		return SchedulerMetrics{}, fmt.Errorf("kube-scheduler pod is not registered. Skipping Scheduler's metrics gathering")
 	}
+
+	var err error
+	g.waitForSchedulerReadyOnce.Do(func() {
+		var lastMetricsFetchErr error
+		if metricsWaitErr := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+			_, lastMetricsFetchErr = g.getMetricsFromPod(g.client, metricsProxyPod, metav1.NamespaceSystem, kubeSchedulerPort)
```

Comment:
In addition to recording the error, should we also record the output from this call, instead of calling getMetricsFromPod again below?

Reply:
Good catch. I will update it.

Reply:
I updated GrabFromControllerManager below as well.